frontend/front/src/components/SurveyComponent/SurveyComponent.js

```diff
@@ -250,17 +252,17 @@ const SurveyComponent = ({
 
 // makes sure data fetching happens BEFORE the form is loaded, so that the form hook can be initialized with the correct default data
 const SurveyComponentWrapper = forwardRef((props, ref) => {
-  const { defaultData, surveyId, style, activeHome } = props;
+  const { defaultData, style, activeHome } = props;
 
   const {
     data: surveyStructure,
     isError: isSurveyError,
     isLoading: isSurveyLoading,
-  } = useGetSurveyStructureQuery(surveyId);
+  } = useGetSurveyStructureQuery();
 
   const formDefault = useMemo(() => {
     if (defaultData) {
-      return defaultData;
+      return buildDataFromSurveyAnswers(defaultData);
```

Comment:
I think you should call this one level up, when you pass the data into this from `SurveyVisitProfile`. `defaultData` is already passed in in the correct format on other pages (at least it should be, we need more tests...) so we should assume that it's always in the format we're looking for by the time it gets here.

Reply:
Done. Pushed up. I put in a useMemo. I have never used this before and I still need to do more reading on it. Just let me know if wrong.